core/state/journal.go

```diff
@@ -87,6 +90,15 @@ func (ch suicideChange) undo(s *StateDB) {
 	}
 }
 
+var ripemd = common.HexToAddress("0000000000000000000000000000000000000003")
+
+func (ch touchChange) undo(s *StateDB) {
+	if !ch.prev && *ch.account != ripemd {
```

Comment:
Good question. If I have this right... :

The `00...003` was one of the accounts touched by the attacker, and also by the statesweep. When it was touched by the statesweep, the actual 'touch', in the form of a `CALL`, went OOG. 

This was at block 2675119 , at this [tx](https://etherscan.io/tx/0xcf416c536ec1a19ed1fb89e4ec7ffb3cf73aa413b3aa9b77d60e4fd81a4296ba). When the geth-bug was fixed initially, it was discovered that the Parity implementation was also flawed; the OOG actually did not revert the deletion of `0..3` (ripemd). 

So geth failed to revert on OOG-transactions, and Parity failed to revert OOG sub-calls, basically. And the fix above is a hack to sync up with parity.